medmodels/_medmodels.pyi

```diff
@@ -241,170 +234,430 @@ class PyMedRecord:
-    def less_or_equal(
-        self, operand: Union[ValueOperand, PyEdgeAttributeOperand]
-    ) -> PyEdgeOperation: ...
-    def equal(
-        self, operand: Union[ValueOperand, PyEdgeAttributeOperand]
-    ) -> PyEdgeOperation: ...
-    def not_equal(
-        self, operand: Union[ValueOperand, PyEdgeAttributeOperand]
-    ) -> PyEdgeOperation: ...
-    def is_in(self, operands: List[MedRecordValue]) -> PyEdgeOperation: ...
-    def not_in(self, operands: List[MedRecordValue]) -> PyEdgeOperation: ...
-    def starts_with(
-        self, operand: Union[ValueOperand, PyEdgeAttributeOperand]
-    ) -> PyEdgeOperation: ...
-    def ends_with(
-        self, operand: Union[ValueOperand, PyEdgeAttributeOperand]
-    ) -> PyEdgeOperation: ...
-    def contains(
-        self, operand: Union[ValueOperand, PyEdgeAttributeOperand]
-    ) -> PyEdgeOperation: ...
-    def add(self, value: MedRecordValue) -> ValueOperand: ...
-    def sub(self, value: MedRecordValue) -> ValueOperand: ...
-    def mul(self, value: MedRecordValue) -> ValueOperand: ...
-    def div(self, value: MedRecordValue) -> ValueOperand: ...
-    def pow(self, value: MedRecordValue) -> ValueOperand: ...
-    def mod(self, value: MedRecordValue) -> ValueOperand: ...
-    def round(self) -> ValueOperand: ...
-    def ceil(self) -> ValueOperand: ...
-    def floor(self) -> ValueOperand: ...
-    def abs(self) -> ValueOperand: ...
-    def sqrt(self) -> ValueOperand: ...
-    def trim(self) -> ValueOperand: ...
-    def trim_start(self) -> ValueOperand: ...
-    def trim_end(self) -> ValueOperand: ...
-    def lowercase(self) -> ValueOperand: ...
-    def uppercase(self) -> ValueOperand: ...
-    def slice(self, start: int, end: int) -> ValueOperand: ...
+    def select_nodes(
+        self, query: Callable[[PyNodeOperand], None]
+    ) -> List[NodeIndex]: ...
+    def select_edges(
+        self, query: Callable[[PyNodeOperand], None]
+    ) -> List[EdgeIndex]: ...
+
+class PyEdgeDirection(Enum):
+    Incoming = 0
+    Outgoing = 1
+    Both = 2
+
+class PyNodeOperand:
+    def attribute(self, attribute: MedRecordAttribute) -> PyMultipleValuesOperand: ...
+    def attributes(self) -> PyAttributesTreeOperand: ...
+    def index(self) -> PyNodeIndicesOperand: ...
+    def in_group(self, group: Union[Group, List[Group]]) -> None: ...
+    def has_attribute(
+        self, attribute: Union[MedRecordAttribute, List[MedRecordAttribute]]
+    ) -> None: ...
+    def outgoing_edges(self) -> PyEdgeOperand: ...
+    def incoming_edges(self) -> PyEdgeOperand: ...
+    def neighbors(self, direction: PyEdgeDirection) -> PyNodeOperand: ...
+    def either_or(
+        self,
+        either: Callable[[PyNodeOperand], None],
+        or_: Callable[[PyNodeOperand], None],
+    ) -> None: ...
+    def deep_clone(self) -> PyNodeOperand: ...
+
+PyNodeIndexComparisonOperand: TypeAlias = Union[NodeIndex, PyNodeIndexOperand]
+PyNodeIndexArithmeticOperand: TypeAlias = PyNodeIndexComparisonOperand
```

Comment:
because it would be a bit weird for the typing in the arithmetic functions to be e.g. `def add(self, index: PyNodeIndexComparisonOperand) -> None` because this is a arithmetic operation, not a comparison operation so I created a TypeAlias. But I can change it back

Reply:
Na, it makes sense. Would it be possible to just have one PyNodeIndexOperand?

Reply:
There already is a `PyNodeIndexOperand`, so no :D
The `PyNodeIndexComparisonOperand` (and `PyNodeIndexArithmeticOperand`) represents that you can pass a `NodeIndex` (`str` or `int`) directly or a `PyNodeIndexOperand`. That's why there needs to be a different name.
And it is used very often, that's why I created a `TypeAlias` instead of using `Union[NodeIndex, PyNodeIndexOperand]` everywhere